PhysicsTools/JetMCAlgos/plugins/HadronAndPartonSelector.cc

```diff
@@ -249,12 +251,20 @@ HadronAndPartonSelector::produce(edm::Event& iEvent, const edm::EventSetup& iSet
    }
 
    // select partons
-   if ( partonMode_!="Undefined" )
+   if ( partonMode_!="Undefined" ) {
      partonSelector_->run(particles,partons);
+     for(reco::GenParticleCollection::const_iterator it = particles->begin(); it != particles->end(); ++it)
+     {
+       if( !(it->status()==3 || (( partonMode_=="Pythia8" ) && (it->status()==23)))) continue;
```

Comment:
Make sure https://github.com/ahinzmann/cmssw/blob/partonFlavourMiniAOD/PhysicsTools/PatAlgos/python/slimming/prunedGenParticles_cfi.py is configured to keep these partons. Also, since we are now changing the meaning of the `partonFlavour()` method, we could completely drop https://github.com/ahinzmann/cmssw/blob/partonFlavourMiniAOD/PhysicsTools/PatAlgos/python/slimming/prunedGenParticles_cfi.py#L21 and save some space that way.